src/flows/pipes/QueryBuilder/readOnly.tsx

```diff
@@ -37,7 +37,9 @@ const QueryBuilder: FC<PipeProp> = ({Context}) => {
                         icon={IconFont.BucketSolid}
                         status={ComponentStatus.Disabled}
                       >
-                        {data.buckets[0].name ?? 'No Bucket Selected'}
+                        {!!data.buckets.length && data.buckets[0].name
+                          ? data.buckets[0].name
+                          : 'No Bucket Selected'}
```

Comment:
```suggestion
                        {!!data?.buckets?.length && data?.buckets?.[0]?.name
                          ? data.buckets[0].name
                          : 'No Bucket Selected'}
```

Reply:
I'm pretty sure `data` and `data.buckets` will always exist...but sure better to be safe than sorry :)